### PR TITLE
removes Makefile and related content

### PR DESCRIPTION
**Pull request by scopatz**
This removes the top-level makefile as it was confusing potential developers, as in #689.  A lot of this could be folded in to the release script, if needed. See also #586

### CODE REVIEW

**Comment by scopatz** [comment]
This one is relatively small :)

File: docs/devguide.rst [hunk 1]
```diff
@@ -304,9 +280,13 @@ When releasing xonsh, make sure to do the following items in order:
 Maintenance Tasks
 --------------------
 You can cleanup your local repository of transient files such as \*.pyc files
-created by unit testing by running:
+created by unit testing by running::
 
-    $ make clean
+    $ rm -f xonsh/lexer_table.py xonsh/parser_table.py
```
Comment (gforsyth) [inline docs/devguide.rst]: Should there be a `run_tests.xsh clean` option to take care of this?

File: scripts/run_tests.xsh [hunk 2]
```diff
@@ -19,11 +19,13 @@ if not os.path.isdir('.git'):
 
 if len($ARGS) == 1:
     # Run all tests.
-    $[make build-tables] # ensure lexer/parser table module is up to date
+    # ensure lexer/parser table module is up to date
+    $[python3 -c 'import setup; setup.build_tables()']
     $[env XONSHRC='' nosetests]
```
Comment (gforsyth) [inline scripts/run_tests.xsh]: Should we suppress the dots when doing this?

File: docs/devguide.rst [hunk 3]
```diff
@@ -78,40 +78,20 @@ First, install nose: http://nose.readthedocs.org/en/latest/. Second, ensure
...
 
 If you're working on a change and haven't yet committed it you can run the
 tests associated with the change. This does not require that the change
 include the unit test module. This will execute any unit tests that are
 part of the change as well as the unit tests for xonsh source modules in
-the change:
+the changes::
 
-    $ make test
+    $ scripts/run_tests.xsh $(which)
```
Comment (gforsyth) [inline docs/devguide.rst]: `which` what?

File: docs/devguide.rst (hunk 1 above)
Comment (scopatz) [inline docs/devguide.rst]: probably, that could be a separate issue though.

File: scripts/run_tests.xsh (hunk 2 above)
Comment (scopatz) [inline scripts/run_tests.xsh]: yes. adding.

**Comment by scopatz** [comment]
I have updated, addressing @gforsyth's major points.

**Comment by gforsyth** [comment]
Hey @scopatz, this looks good to go but the binstar CI is still pending?  It's been... 17 hours?  That seems silly.

**Comment by scopatz** [comment]
Yeah, @gforsyth, I bet that this is an anaconda ci issue. It is probably OK to go ahead and merge since it tested just fine on the other systems.

**Comment by gforsyth** [comment]
Thanks @scopatz!  I also learned that you can't merge PRs with test failures from your phone.

**Comment by scopatz** [comment]
Who would have thought? ;)

**Pull request merged**